storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3MultiPartOutputStream.java

```diff
@@ -56,7 +55,7 @@ public class S3MultiPartOutputStream extends OutputStream {
     private final String uploadId;
     private final List<PartETag> partETags = new ArrayList<>();
 
-    private boolean closed;
+    private volatile boolean closed;
```

Comment:
Should we do this? This class is not thread-safe and is supposed to be used inside one thread anyway.

Reply:
Got it. Was getting worried as this value changes in multiple places.
Will add a note about this to the stream, and I'm extending tests to validate this value.